diff --git a/drivers/net/wireless/rtl8723bu/Kconfig b/drivers/net/wireless/rtl8723bu/Kconfig
new file mode 100644
index 000000000..63bbe8e63
--- /dev/null
+++ b/drivers/net/wireless/rtl8723bu/Kconfig
@@ -0,0 +1,4 @@
+config RTL8723BU
+	tristate "Realtek 8723B USB WiFi"
+	help
+	  Help message of RTL8723BU
diff --git a/drivers/net/wireless/rtl8723bu/Makefile b/drivers/net/wireless/rtl8723bu/Makefile
index 9de45eddd..172f597e1 100644
--- a/drivers/net/wireless/rtl8723bu/Makefile
+++ b/drivers/net/wireless/rtl8723bu/Makefile
@@ -25,7 +25,7 @@ CONFIG_PLATFORM_FS_MX61 = n
 
 CONFIG_DRVEXT_MODULE = n
 
-export TopDIR ?= $(shell pwd)
+export TopDIR ?= $(src)
 
 ########### COMMON  #################################
 HCI_NAME = usb
diff --git a/drivers/net/wireless/rtl8723bu/Makefile b/drivers/net/wireless/rtl8723bu/Makefile
index 032cbb3..e753323 100644
--- a/drivers/net/wireless/rtl8723bu/Makefile
+++ b/drivers/net/wireless/rtl8723bu/Makefile
@@ -18,7 +18,7 @@ EXTRA_CFLAGS += -Wno-unused-label
 EXTRA_CFLAGS += -Wno-unused-parameter
 EXTRA_CFLAGS += -Wno-unused-function
 EXTRA_CFLAGS += -Wno-unused
-EXTRA_CFLAGS += -DCONFIG_CONCURRENT_MODE
+#EXTRA_CFLAGS += -DCONFIG_CONCURRENT_MODE
 
 ccflags-y += -D__CHECK_ENDIAN__
 
diff --git a/drivers/net/wireless/rtl8723bu/include/autoconf.h b/drivers/net/wireless/rtl8723bu/include/autoconf.h
index d5556af..6fb6fa9 100644
--- a/drivers/net/wireless/rtl8723bu/include/autoconf.h
+++ b/drivers/net/wireless/rtl8723bu/include/autoconf.h
@@ -71,7 +71,7 @@
 #endif
 
 
-#define CONFIG_IPS	1
+//#define CONFIG_IPS	1
 #ifdef CONFIG_IPS
 	//#define CONFIG_IPS_LEVEL_2	1 //enable this to set default IPS mode to IPS_LEVEL_2
 #endif
 
