From 3c1d703b5da9a9222ac12cd0007c0d82c0153663 Mon Sep 17 00:00:00 2001
From: Mikalai Ramanovich <mikalair@outlook.com>
Date: Thu, 12 Nov 2020 00:35:21 +0300
Subject: [PATCH] Fix compilation

---
 CyReP/RiotCrypt.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/external/RIoT/CyReP/RiotCrypt.c b/external/RIoT/CyReP/RiotCrypt.c
index bd01ee1..4743c1b 100644
--- a/external/RIoT/CyReP/RiotCrypt.c
+++ b/external/RIoT/CyReP/RiotCrypt.c
@@ -417,10 +417,6 @@ RiotCrypt_DERDecodeECCSignature(
     uint32_t                encSize = ((BIGLEN - 1) * 4);
     uint32_t                seqSize;
 
-    if (DerSize > UINT32_MAX) {
-        return RIOT_BAD_FORMAT;
-    }
-
     DERInitDecoder(&ddc, DerSig, (uint32_t)DerSize);
     if (DERGetSequenceOrSetLength(&ddc, true, &seqSize) < 0) {
         return RIOT_BAD_FORMAT;
@@ -437,4 +433,4 @@ RiotCrypt_DERDecodeECCSignature(
     BigIntToBigVal(&Sig->s, encBuffer, seqSize);
 
     return RIOT_SUCCESS;
-}
\ No newline at end of file
+}
-- 
2.29.2

